src/test/kotlin/org/jetbrains/research/ml/ast/util/FileTestUtil.kt

```diff
@@ -2,45 +2,35 @@ package org.jetbrains.research.ml.ast.util
 
 import java.io.File
 
+enum class Extension(val value: String) {
+    Py(".py"), Xml(".xml")
+}
+
 object FileTestUtil {
 
     val File.content: String
         get() = this.readText().removeSuffix("\n")
 
     fun getInAndOutFilesMap(
         folder: String,
-        oldExtension: String = ".py",
-        newExtension: String = ".py"
+        inExt: Extension = Extension.Py,
+        outExt: Extension = Extension.Py
     ): Map<File, File> {
-        val inFileRegEx = "in_\\d*$oldExtension".toRegex()
-        val inOutFileRegEx: Regex = "(in|out)_\\d*($oldExtension|$newExtension)".toRegex()
-        val (inFiles, outFiles) = getNestedFiles(folder).toList().filter { inOutFileRegEx.containsMatchIn(it.name) }
-            .partition { inFileRegEx.containsMatchIn(it.name) }
-        if (inFiles.size != outFiles.size) {
-            throw IllegalArgumentException(
-                "Size of the list of input files does not equal size of the list of output files if the folder: $folder"
-            )
-        }
-        return inFiles.associateWith { inFile ->
-            // TODO: can I do it better?
-            val outFileName = inFile.name.replace("in", "out").replace(oldExtension, newExtension)
-            val outFile = File("${inFile.parent}/$outFileName")
-            if (!outFile.exists()) {
-                throw IllegalArgumentException("Out file $outFile does not exist!")
-            }
-            outFile
-        }
-    }
+        fun File.number(): Int = "(?<=(in|out)_)\\d+".toRegex().find(this.name)!!.value.toInt()
+        fun File.isInFile(): Boolean = "in_\\d+.*${inExt.value}".toRegex().containsMatchIn(this.name)
+        fun File.isOutFile(): Boolean = "out_\\d+.*${outExt.value}".toRegex().containsMatchIn(this.name)
 
-    private fun getNestedFiles(directoryName: String, files: MutableList<File> = ArrayList()): Sequence<File> {
-        val root = File(directoryName)
-        root.listFiles()?.forEach {
-            if (it.isFile) {
-                files.add(it)
-            } else if (it.isDirectory) {
-                getNestedFiles(it.absolutePath, files)
-            }
-        }
-        return files.asSequence()
+        val (files, folders) = File(folder).listFiles().orEmpty().partition { it.isFile }
+//      Process files in the given folder
+        val inAndOutFiles = files.filter { it.isInFile() || it.isOutFile() }.groupBy { it.number() }
+        val inAndOutFilesMap = inAndOutFiles.toSortedMap().map { (number, files) ->
+            require(files.size == 2) { "There are less or more than 2 test files with number $number" }
+            val (inFile, outFile) = files.sorted().zipWithNext().first()
```

Comment:
Relying on `in` alphabetically preceding `out` seems a little fragile to me :thinking: (maybe it's fine since it's working)

Reply:
I think we should not rely on alphabetical order. Especially, if we add `inPrefix` and `outPrefix` variables.